anoma/namada-interface
# chore: bump Namada to 0.40.0

This PR bumps the version of `Namada` SDK to `v0.40.0`, and makes the necessary changes

- [x] Bump `@namada/shared` to `v0.40.0` of Namada
- [x] Fix types and `Transfer` args (these have changed!) in `shared`
- [x] Update our SDK package to reflect changes in `Transfer` args, regenerate docs
- [x] Update our schema to properly deserialize `Transfer` type to reflect Namada changes
- [x] Small change to `fetchWasmHashes` to ensure that the check works in the event of an RPC error (interface was adding hashes as an empty array when it shouldn't have been invoked at all, due to a local CORS error)
- [x] Regenerate `packages/types` docs

### Testing

I attempted to test this against Luminare, which is still on `v0.39.0`, and it did not work.

- Set up a local chain based on `v0.40.0` with balances, and point a local indexer to it (I reset the DB as well)
- Set up the extension with an account containing balances
- Issue Tx, and they should succeed!
